base/promotion.jl

```diff
@@ -258,6 +258,17 @@ function tailjoin(A::SimpleVector, i::Int)
     return t
 end
 
+## type equality
+
+function _types_are_equal(A::Type, B::Type)
+    @_total_meta
+    ccall(:jl_types_equal, Cint, (Any, Any), A, B) !== Cint(0)
+end
+
+function _type_is_bottom(X::Type)
+    X <: Bottom
```

Comment:
why `<:` rather than `===` ?

Reply:
I think you're right, or at least I can't find a counterexample. Replacing with `X === Bottom`.